Applications/spire/source/time_and_sales/time_and_sales_properties_dialog.cpp

```diff
@@ -206,95 +207,18 @@ time_and_sales_properties_dialog::time_and_sales_properties_dialog(
-  auto reset_default_button = new flat_button(tr("Reset Default"), this);
-  reset_default_button->connect_clicked_signal(
-    [=] { m_save_default_signal(time_and_sales_properties()); });
-  reset_default_button->setFont(generic_button_font);
-  reset_default_button->set_style(generic_button_default_style);
-  reset_default_button->set_hover_style(generic_button_hover_style);
-  reset_default_button->set_focus_style(generic_button_focused_style);
-  buttons_layout_2->addWidget(reset_default_button);
-  buttons_layout_2->setStretchFactor(reset_default_button, 26);
-  buttons_layout->addLayout(buttons_layout_2);
-  buttons_layout->setStretchFactor(buttons_layout_2, 100);
-  buttons_layout->addStretch(60);
-  auto buttons_layout_3 = new QVBoxLayout();
-  buttons_layout_3->setContentsMargins({});
-  buttons_layout_3->setSpacing(0);
-  auto apply_to_all_button = new flat_button(tr("Apply To All"), this);
-  apply_to_all_button->connect_clicked_signal(
+  buttons_widget->connect_apply_to_all_signal(
     [=] { m_apply_all_signal(get_properties()); });
-  apply_to_all_button->setFont(generic_button_font);
-  apply_to_all_button->set_style(generic_button_default_style);
-  apply_to_all_button->set_hover_style(generic_button_hover_style);
-  apply_to_all_button->set_focus_style(generic_button_focused_style);
-  buttons_layout_3->addWidget(apply_to_all_button);
-  buttons_layout_3->setStretchFactor(apply_to_all_button, 26);
-  buttons_layout_3->addStretch(8);
-  auto cancel_button = new flat_button(tr("Cancel"), this);
-  cancel_button->connect_clicked_signal(
-    [=] { reject(); });
-  cancel_button->setFont(generic_button_font);
-  cancel_button->set_style(generic_button_default_style);
-  cancel_button->set_hover_style(generic_button_hover_style);
-  cancel_button->set_focus_style(generic_button_focused_style);
-  buttons_layout_3->addWidget(cancel_button);
-  buttons_layout_3->setStretchFactor(cancel_button, 26);
-  buttons_layout->addLayout(buttons_layout_3);
-  buttons_layout->setStretchFactor(buttons_layout_3, 100);
-  buttons_layout->addStretch(8);
-  auto buttons_layout_4 = new QVBoxLayout();
-  buttons_layout_4->setContentsMargins({});
-  buttons_layout_4->setSpacing(0);
-  auto apply_button = new flat_button(tr("Apply"), this);
-  setTabOrder(apply_to_all_button, apply_button);
-  apply_button->connect_clicked_signal(
+  buttons_widget->connect_apply_signal(
     [=] { m_apply_signal(get_properties()); });
-  apply_button->setFont(generic_button_font);
-  apply_button->set_style(generic_button_default_style);
-  apply_button->set_hover_style(generic_button_hover_style);
-  apply_button->set_focus_style(generic_button_focused_style);
-  buttons_layout_4->addWidget(apply_button);
-  buttons_layout_4->setStretchFactor(apply_button, 26);
-  buttons_layout_4->addStretch(8);
-  auto ok_button = new flat_button(tr("OK"), this);
-  ok_button->connect_clicked_signal(
+  buttons_widget->connect_ok_signal(
     [=] { accept(); });
```

Comment:
Fit this on the line above as well. Only continue a line if there's no reasonable way to fit it in 80 columns.